cmd/dockerd/malformed_host_override.go

```diff
@@ -0,0 +1,80 @@
+// +build !windows
+
+package main
```

Comment:
It'd be great if this was in a separate package for testing purposes.
Also tests/benchmarks would be helpful.

I have some benchmarks pending.